### ISSUE TITLE
10.10.1  >  Dual Monitor  >  2nd monitor Fullscreen stuck

### ISSUE THREAD

**Issue by bwv656** [eczarny/spectacle#303]
First of all thank you so much for this awesome app! I find it extremely useful!!
I am using 10.0.1 with Mission Control > [  ] Displays have separate spaces  (Unchecked).
On my Primary display with the menubar, I have no problem using Spectacle. It works beautifully.

However on my 2nd monitor on the right (without the menubar), whenever I make any windows full screen, it just stays there, it doesn't respond to any key commands from that point.

So I tried Mission Control > [✓] Displays have separate spaces  (checked)
In this setting menubars are on both screen. And Spectacle works on both screen.
But I prefer to have ' Displays have separate spaces (unchecked)'

So in short it seems like the issue is with the display without the menubar.
It would be great if you can look into this soon.
Thank you so much!!

-j

**Comment by pklinef** [comment]
I use the same configuration and have the same issue.  After using fullscreen on the second monitor, I have to move the window before I can use Spectacle to reposition the window again.  Fullscreen works on the primary external monitor and the laptop screen but not the second Apple Thunderbolt display.  

Thank you for this awesome open source application.  Let me know if there is some way I can help debug this issue.

**Comment by eczarny** [comment]
Do you mean when the windows are in OS X's fullscreen mode? Technically Spectacle should not allow you to move a window in fullscreen mode, that is actually a defect I need to fix. See #296.

**Comment by WoLpH** [comment]
The problem occurs after maximizing a screen `⌥⌘F`.

To reproduce:

1. Move a window to the secondary monitor
2. `⌥⌘F` to maximize (perhaps this button should be renamed in Spectacle, it's called fullscreen which it isn't)
3. `⌥⌘←` (or any other tiling spectacle command) will be ignored

My workaround right now:

1. `⇧⌥⌘←` to move the window to the main monitor
2. `⌥⌘←` resize the screen
3. `⇧⌥⌘ → ` move the screen back to the secondary monitor
4. `⌥⌘←` resize the screen again (since moving it to the other monitor doesn't save the placement, another bug?)

**Comment by eczarny** [comment]
is this still an issue in 0.8.10?

**Comment by WoLpH** [comment]
Sorry, didn't see your message earlier. But yet, the problem is still there. After maximizing I can't modify the window sizes anymore. After dragging the screen a few pixels from it's original position it works again though.

Let me know if I can help debug it, I have Xcode available but not entirely sure where to start :)

**Comment by WoLpH** [comment]
Slightly more info here, it appears it doesn't work with some apps. With iTerm2 I have no problems, but with Chrome it's broken.

**Comment by WoLpH** [comment]
After running spectacle in the debugger for a bit I found out that it's this line that's causing the problem:

From `SpectacleWindowPositionManager.m in `moveFrontMostWindowWithAction`:
```objective-c
    if (frontMostWindowElement.isSheet || CGRectIsNull(frontMostWindowRect) || CGRectIsNull(frameOfScreen) || CGRectIsNull(visibleFrameOfScreen) || CGRectEqualToRect(frontMostWindowRect, frameOfScreen)) {
        NSBeep();
        
        return;
    }
```

Specifically, this boolean is `true`: `CGRectEqualToRect(frontMostWindowRect, frameOfScreen)`

Simply commenting that if statement makes everything work as expected, although I'm sure it has a reason to be there :P

**Comment by WoLpH** [comment]
@eczarny Would you have some time to take a look?

I've been using a version with that line commented out without any issues for about a month now :)

**Comment by eczarny** [comment]
Sorry it took so long to get back to this issue! Let me look through why that condition was added to that check and ensure the fix will fully work as expected. I won't have access to another display until next week so it may take a bit more time to look into this.

**Comment by eczarny** [comment]
The `CGRectEqualToRect(frontMostWindowRect, frameOfScreen)` check was added to try and prevent Spectacle from interacting with fullscreen apps. Unfortunately this only works on the main display (the display OS X considers the main screen, or the display where the menu bar is displayed) and breaks with any other display.

Spectacle currently doesn't completely play nicely with fullscreen apps (see #296) so this check is safe to remove. I'll need to find a better way to check that an app is fullscreen or not.
